lib/metrics.ts

```diff
@@ -1,74 +1,83 @@
-'use strict';
+import promBundle from "express-prom-bundle";
+import { ILogger } from "nchat-dev-common";
 
-const promBundle = require('express-prom-bundle');
+import { PrometheusMetricsBackend } from "./backends/prometheus";
 
-const { PrometheusMetricsBackend } = require('./backends/prometheus');
+import { ExpressMiddlewareDefaultSettings, DefaultBackendSettings } from "./constants";
+import { IBackendSettings, IMetricsBackend } from "./types";
 
-const { ExpressMiddlewareDefaultSettings, DefaultBackendSettings } = require('./constants');
+export class Metrics {
+    private _logger: ILogger;
+    private _backend: IMetricsBackend;
+    private _expressMiddlewareProvider: any;
+    private _expressMiddlewareSettings: any;
+    private _prometheusMiddleware: any;
```

Comment:
```suggestion
    private _expressMiddlewareProvider: ExpressMiddlewareProvider;
    private _expressMiddlewareSettings: IExpressMiddlewareSettings;
    private _prometheusMiddleware?: RequestHandler;
```

export interface IExpressMiddlewareSettings {
     // https://github.com/jochen-schweizer/express-prom-bundle/blob/master/types/index.d.ts#L20
}

export type ExpressMiddlewareProvider = (settings: IExpressMiddlewareSettings) => RequestHandler;